src/v/serde/rw/vector.h

```diff
@@ -35,6 +35,7 @@ concept Vector = requires(T t) {
     t.begin();
     t.end();
     { t.size() } -> std::convertible_to<std::size_t>;
+    t.shrink_to_fit();
```

Comment:
the commit message doesn't seem to indicate that this was a problem prior to the commit, so i'm wondering: is this related to the previous commit that added shrink_to_fit to the bytes type? if so, we may have an issue.

Reply:
Yes it's related.  I'm storing the sha256 checksum as bytes into the kvstore.  I can switch to use fragmented_vector if that's an issue.

Reply:
>Yes it's related. I'm storing the sha256 checksum as bytes into the kvstore. I can switch to use fragmented_vector if that's an issue.

but `bytes` should not be being handled by the vector serde encoder. i think the issue is that the serde encoder is working with "things that look like vectors". i think can drop the shrink-to-fit and the rw/vector.h and include rw/bytes.h?

separately ill take a todo item to make this stricter, or maybe they should be combined if their on-wire formats are identical.

Reply:
> include rw/bytes.h?

oh my bad, yeah good call

Reply:
I will say that the change in `rw/vector.h` I think is still good (though probably no longer relevant to this PR), since the `tag_invoke(tag_t<read_tag>)` function calls `shrink_to_fit()`